fiaas_skipper/deploy/channel.py

```diff
@@ -23,14 +20,26 @@ def __init__(self, baseurl):
     def __call__(self, name, tag):
         r = requests.get('%s/%s/%s.json' % (self._baseurl, name, tag))
         LOG.debug('Retrieving meta data for %s/%s/%s.json' % (self._baseurl, name, tag))
-        return ReleaseChannel(name, tag, metadata=r.json())
+        metadata = r.json()
+        spec = self._get_spec(metadata.get('spec', None))
+        return ReleaseChannel(name, tag, metadata=metadata, spec=spec)
+
+    @staticmethod
+    def _get_spec(url):
+        """Load spec file yaml from url"""
+        if not url:
+            LOG.error("No spec url specified")
+            raise ValueError("Channel metadata contained no config URL")
+        LOG.debug("Loading spec from channel metadata: " + url)
+        return requests.get(url).text
```

Comment:
We should probably add add a `raise_for_status` call here to avoid strange failures from handling unexpected content from a 4xx or 5xx response later